### ISSUE TITLE
Questions regarding your implementation

### ISSUE THREAD

**Issue by NIRVANALAN** [shchur/overlapping-community-detection#2]
Hi, I have read yoru code and have some questions here:

1. you adopted the self-implemented ```l2_reg_loss``` and add it to BP loss. Why don't you use the l2_reg of ```Adam``` optimizer by setting ```weight_decay``` param?
2. in your gcn implementation, I noticed
![image](https://user-images.githubusercontent.com/23359323/74533166-e3ca4380-4f6b-11ea-9138-4848487a3700.png)
I wonder why are there two dropout in one layer?

**Comment by shchur** [comment]
Hi,

1. I'm not sure if that's the right thing to do, but I remember reading that Pytorch and Tensorflow implement L2 regularization / weight decay differently (https://openreview.net/pdf?id=rk6qdGgCZ). So I decided to use the same version of L2 regularization as in the original Tensorflow implementation for consistency.

2. That indeed seems to be a mistake, thanks a lot for pointing it out! I will have a look at it on Monday.

**Comment by shchur** [comment]
I have just fixed this issue in 66a015d4632e62aa9a6b4e75ac018253734ff5f6, everything seems to work fine now. Thanks again for point it out!
